py_hamt/sharded_zarr_store.py

```
@@ -0,0 +1,870 @@
+                            await self.resize_store(new_shape=tuple(new_array_shape))
+                        finally:
+                            # All waiting tasks will now un-pause and proceed safely.
+                            self._resize_complete.set()
+
+        raw_data_bytes = value.to_bytes()
+        # Save the data to CAS first to get its CID.
+        # Metadata is often saved as 'raw', chunks as well unless compressed.
+        try:
+            data_cid_obj = await self.cas.save(raw_data_bytes, codec="raw")
+            await self.set_pointer(key, str(data_cid_obj))
+        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)